internal/testrunner/runners/system/runner.go

```diff
@@ -747,6 +759,43 @@ func validateFields(docs []common.MapStr, fieldsValidator *fields.Validator, dat
 	return nil
 }
 
+func getDocEventCount(events map[string]interface{}) int {
+	for k, v := range events {
+		if reflect.ValueOf(v).Kind() == reflect.Slice {
+			e := events[k].([]interface{})
+			logger.Debugf("found %d events under \"%s\"", len(e), k)
+			return len(e)
+		}
+	}
+	return 0
+}
+
+func getEventCount(docs []common.MapStr) (lenEvents int, err error) {
```

Comment:
It doesn't look obvious how events are counted here, it could be just `len(docs)`, but this is also looking for events in the `message`. Could you add a comment about this here?